server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java

```diff
@@ -284,4 +304,320 @@ public void setMessagesShouldDeleteMatchingMessagesWhenMixed() throws Exception
             .body(NAME, equalTo("messages"))
             .body(ARGUMENTS + ".list", hasSize(1));
     }
-}
+
+    @Test
+    public void setMessagesShouldReturnUpdatedIdAndNoErrorWhenIsUnreadPassedToFalse() throws MailboxException {
+        // Given
+        jmapServer.serverProbe().createMailbox(MailboxConstants.USER_NAMESPACE, username, "mailbox");
+
+        jmapServer.serverProbe().appendMessage(username, new MailboxPath(MailboxConstants.USER_NAMESPACE, username, "mailbox"),
+                new ByteArrayInputStream("Subject: test\r\n\r\ntestmail".getBytes()), new Date(), false, new Flags());
+        embeddedElasticSearch.awaitForElasticSearch();
+
+        String presumedMessageId = username + "|mailbox|1";
+
+        // When
+        given()
+            .accept(ContentType.JSON)
+            .contentType(ContentType.JSON)
+            .header("Authorization", accessToken.serialize())
+            .body(String.format("[[\"setMessages\", {\"update\": {\"%s\" : { \"isUnread\" : false } } }, \"#0\"]]", presumedMessageId))
+        .when()
+            .post("/jmap")
+        // Then
+        .then()
+            .spec(getSetMessagesUpdateOKResponseAssertions(presumedMessageId))
+            .log().ifValidationFails();
+    }
+
+    private ResponseSpecification getSetMessagesUpdateOKResponseAssertions(String messageId) {
+        ResponseSpecBuilder builder = new ResponseSpecBuilder()
+                .expectStatusCode(200)
+                .expectBody(NAME, equalTo("messagesSet"))
+                .expectBody(ARGUMENTS + ".updated", hasSize(1))
+                .expectBody(ARGUMENTS + ".updated", contains(messageId))
+                .expectBody(ARGUMENTS + ".error", isEmptyOrNullString())
+                .expectBody(ARGUMENTS + ".notUpdated", not(hasKey(messageId)));
+        return builder.build();
+    }
+
+    @Test
+    public void setMessagesShouldMarkAsReadWhenIsUnreadPassedToFalse() throws MailboxException {
+        // Given
+        jmapServer.serverProbe().createMailbox(MailboxConstants.USER_NAMESPACE, username, "mailbox");
+
+        jmapServer.serverProbe().appendMessage(username, new MailboxPath(MailboxConstants.USER_NAMESPACE, username, "mailbox"),
+                new ByteArrayInputStream("Subject: test\r\n\r\ntestmail".getBytes()), new Date(), false, new Flags());
+        embeddedElasticSearch.awaitForElasticSearch();
+
+        String presumedMessageId = username + "|mailbox|1";
+        with()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+                .header("Authorization", accessToken.serialize())
+                .body(String.format("[[\"setMessages\", {\"update\": {\"%s\" : { \"isUnread\" : false } } }, \"#0\"]]", presumedMessageId))
+        // When
+                .post("/jmap");
+        // Then
+        with()
```

Comment:
per raphael's request, I've changed all of them previously (removed given/when). Technically they are the same thing and from a semantic point of view, we're only using the rest assured helpers in the second part of the test to build up assertions, not to write scenarii specifications and context.
The comment (//Then) is part of the test writing convention, and the "with" keyword is here to validate/assert on the expected result.